### PR TITLE
Add calculation of code structure complexity

### PR DESCRIPTION
**Pull request by Feyd-Rauth**

### CODE REVIEW

File: pkg/codestruct/codestruct.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+func (m *Metric) ParseNode(n ast.Node) {
+	m.Comp += getCodeStructComp(n, &m.Config)
+}
+
+func getCodeStructComp(n ast.Node, cfg *Weights) uint {
+	comp := uint(0)
+
+	switch n.(type) {
```
Comment (bragov4ik) [inline pkg/codestruct/codestruct.go:25]: Maybe also count functional literals (FuncLit) with their own weight?

File: pkg/codestruct/codestruct.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	case *ast.StructType:
+		comp += cfg.Struct
```
Comment (bragov4ik) [inline pkg/codestruct/codestruct.go:27]: Why use += (weight) here when the result is always 0 + (weight) = (weight)? It seems that just assigning the weight is more readable

File: pkg/codestruct/codestruct.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package codestruct
+
+import (
+	"go/ast"
+)
+
+type Weights struct {
+	Func      uint `xml:"func"`
```
Comment (bragov4ik) [inline pkg/codestruct/codestruct.go:8]: Btw why do we use `uint` as weights everywhere? Won't using `float64` make tweaking them easier (by allowing changing them more preciesly)?

File: pkg/codestruct/codestruct.go (hunk 2 above)
Comment (Feyd-Rauth) [reply pkg/codestruct/codestruct.go:27]: My bad, I'll fix it

**Review by bragov4ik — APPROVED** [review]
Well done!

**Pull request merged**